core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -470,7 +471,7 @@ public EmbeddedCacheManager startCaches(final String... cacheNames) {
       authzHelper.checkPermission(AuthorizationPermission.LIFECYCLE);
       List<Thread> threads = new ArrayList<Thread>(cacheNames.length);
       final AtomicReference<RuntimeException> exception = new AtomicReference<RuntimeException>(null);
-      for (final String cacheName : cacheNames) {
+      for (final String cacheName : new HashSet<String>(Arrays.asList(cacheNames))) {
```

Comment:
Pico-optimisation by uniquefying the cache names. InfinispanDirectoryProvider may be configured so that its caches (data/metdata/locking) are the same, so it will invoke cacheManager.startCaches with repeating cache names. A thread will be spawned for each occurrence. This does not lead to any errors but the extra threads just sit there waiting for the first to finish creating the cache.

Should probably extract that in a different commit.